### ISSUE TITLE
Usage page clarification - build or reporting?

### ISSUE THREAD

**Issue by LIttleAncientForestKami** [mojohaus/taglist-maven-plugin#98]
Hi there! I just had a case to use the plugin (thank you, BTW, as I've used it several times in the past) and spent close to 2 hours trying to incorporate FIXMEs. My memory told me that it was just a tag. I then found something like this on the Usage page:

```
For backwards compatibility with versions prior to the 2.4 release, the legacy tags configuration is supported; however, only exact matching is used with legacy configuration.

    <project>
      ...
            <configuration>
              <tags>
                <tag>TODO</tag>
                <tag>FIXME</tag>
                <tag>@todo</tag>
                <tag>@deprecated</tag>
              </tags>
            </configuration>
      ...
    </project>
```

Small typo: "backwards" -> "backward".
Missing article: "with legacy" "with the legacy"

## True problem

The usage page config doesn't seem to be working. I mean here the legacy config with a new version of the plugin.

I've created a dummy project for this and:
![Taglist_2022-10-29_21-30-52](https://user-images.githubusercontent.com/1167811/198849521-fbeaeb94-5189-4b7c-aceb-ba4e0043241a.png)

In a nutshell:

1. plugin in build, version 3.0.0, config mentions FIXME, `mvn taglist:taglist` does NOT pick up FIXME is the new kid on the block. TODOs are being sought.
2. same with reporting
3. downgrade to 2.3 and the config works.
4. there's no message, other than "using legacy tag" warning. 

What I'd expect:

1. usage page suggests this works. Please either make it so or clarify that version needs to be downgraded for it to work.
2. clearly mark that it's no longer REPORTING but BUILD, unless it's the site command. This I did not tests. I'm however certain, that adding a line or two about BUILD being the target section for `mvn taglist:taglist` would be to overall benefit.

**Comment by slawekjaranowski** [comment]
legacy `tags` was removed in version 3.0.0

 - 0fd1253fe0d6b7b6a8ebfbe98419a61b2df1b41d
 - 422818a3abe900a02430ede33ca901715eecc6eb

documentation should be updated
